code/modules/mob/living/simple_animal/hostile/giant_spider/nurse.dm

```diff
@@ -25,7 +25,7 @@
 			stop_automated_movement = 0
 
 /mob/living/simple_animal/hostile/giant_spider/nurse/proc/check_evolve()
-	if(spider_queens.len < MAX_SQUEENS)
+	if(spider_queens.len < MAX_SQUEENS && !key)	//don't evolve if there's a player inside
```

Comment:
Wouldn't !client or something along those lines be better?

Reply:
I'm not sure what all the differences between the two would be, honestly.

Reply:
I think `key` (or `ckey`) is right here, since `client` is null if the person is logged out